File: Lua/Soutenance/controllers/gameControllers/CollisionController.lua
local CollisionController = {};

function CollisionController:New()
    local collisionController = {};
    setmetatable(collisionController, {__index = CollisionController});

    collisionController.colliders = {};

    table.insert(controllers, collisionController);

    return collisionController;
end

function CollisionController:Update()
    for __, collider in ipairs(self.colliders) do
        for __, other in ipairs(self.colliders) do
            if collider ~= other then
                collider:CheckCollision(other);
            end
        end
    end
end

function CollisionController:DrawColliders()
    for __, collider in ipairs(self.colliders) do
        collider:Draw();
    end
end

return CollisionController;

File: MonoGame/Soutenance_MonoGame/Controllers/CollisionController.cs
using Microsoft.Xna.Framework;
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;
using System.Runtime.Serialization;
using System.Text;
using System.Threading.Tasks;

namespace Soutenance_MonoGame
{
    public static class CollisionController
    {
        public static List<Collider> colliders = new List<Collider>();

        public static void UpdateColliders()
        {
            foreach (Collider col in colliders)
            {
                col.Update();
                foreach (Collider other in colliders)
                {
                    if (col != other)
                    {
                        col.CheckCollision(other);
                    }
                }
            }
        }

        public static void DrawColliders()
        {
            foreach (Collider col in colliders)
            {
                col.Draw();
            }
        }

        public static void CleanColliders()
        {
            for (int i = colliders.Count - 1; i >= 0; i--)
            {
                if (!colliders[i].parent.enabled)
                    colliders.Remove(colliders[i]);
            }
        }
    }
}
